File: src/main/java/land/leets/domain/portfolio/presentation/dto/PortfolioResponse.java
package land.leets.domain.portfolio.presentation.dto;

import land.leets.domain.contributor.domain.Contributor;
import land.leets.domain.portfolio.domain.Portfolio;
import land.leets.domain.portfolio.domain.ProjectScope;
import land.leets.domain.portfolio.domain.ProjectType;
import lombok.AllArgsConstructor;
import lombok.Getter;

import java.time.LocalDate;
import java.util.List;

@Getter
@AllArgsConstructor
public class PortfolioResponse {

    private Long portfolioId;

    private Long generation;

    private String name;

    private String summary;

    private String description;

    private ProjectType type;

    private ProjectScope scope;

    private LocalDate startDate;

    private LocalDate endDate;

    private String serviceUrl;

    private String logoImgName;

    private String mainImgName;

    private List<Contributor> contributors;

    public static PortfolioResponse from(Portfolio portfolio) {
        return new PortfolioResponse(
                portfolio.getPortfolioId(),
                portfolio.getGeneration(),
                portfolio.getName(),
                portfolio.getSummary(),
                portfolio.getDescription(),
                portfolio.getType(),
                portfolio.getScope(),
                portfolio.getStartDate(),
                portfolio.getEndDate(),
                portfolio.getServiceUrl(),
                portfolio.getLogoImgName(),
                portfolio.getMainImgName(),
                portfolio.getContributors()
        );
    }
}

File: src/main/kotlin/land/leets/domain/portfolio/presentation/dto/PortfolioResponse.kt
package land.leets.domain.portfolio.presentation.dto

import land.leets.domain.contributor.domain.Contributor
import land.leets.domain.portfolio.domain.Portfolio
import land.leets.domain.portfolio.type.ProjectScope
import land.leets.domain.portfolio.type.ProjectType
import java.time.LocalDate

data class PortfolioResponse(
    val portfolioId: Long,
    val generation: Long,
    val name: String,
    val summary: String,
    val description: String,
    val type: ProjectType,
    val scope: ProjectScope,
    val startDate: LocalDate,
    val endDate: LocalDate,
    val serviceUrl: String,
    val logoImgName: String,
    val mainImgName: String,
    val contributors: List<Contributor>
) {
    companion object {
        fun from(portfolio: Portfolio): PortfolioResponse {
            return PortfolioResponse(
                portfolioId = portfolio.portfolioId,
                generation = portfolio.generation,
                name = portfolio.name,
                summary = portfolio.summary,
                description = portfolio.description,
                type = portfolio.type,
                scope = portfolio.scope,
                startDate = portfolio.startDate,
                endDate = portfolio.endDate,
                serviceUrl = portfolio.serviceUrl,
                logoImgName = portfolio.logoImgName,
                mainImgName = portfolio.mainImgName,
                contributors = portfolio.contributors
            )
        }
    }
}
